src/include/ceph_fs.h

```diff
@@ -587,10 +587,18 @@ union ceph_mds_request_args {
 	struct {
 		__le32 mode;
 		__le32 rdev;
+		__le32 owner_uid, owner_gid;
 	} __attribute__ ((packed)) mknod;
 	struct {
 		__le32 mode;
+		__le32 owner_uid, owner_gid;
 	} __attribute__ ((packed)) mkdir;
+	struct {
+		__le32 owner_uid, owner_gid;
+	} __attribute__ ((packed)) symlink;
+	struct {
+		__le32 owner_uid, owner_gid;
+	} __attribute__ ((packed)) rename;
```

Comment:
Why not just define this in ` struct ceph_mds_request_head{}` directly just after `caller_{uid,gid}` ? And set them to `-1` as default and only in the `WRITE` requests will they make sense ?

And also please be careful when extending the members in ` struct ceph_mds_request_args{}`, because the kclient will just copy the memories and will always assume the size of the memories are the same between kclient and MDS.

Reply:
My motivation to extend `union ceph_mds_request_args` was to save some bytes :) By adding these two fields to the union I'm not increasing it's size, because a biggest member of union is `setattr` and I'm not extending it.

If you think that it's still better and safer to extend `struct ceph_mds_request_head` I'll do. Just confirm, please.

Reply:
> My motivation to extend `union ceph_mds_request_args` was to save some bytes in network packets :) By adding these two fields to the union I'm not increasing it's size, because a biggest member of union is `setattr` and I'm not extending it.
> 
> Second reason it logical. I'm adding new fields only to those union members, who need them.
> 
> If you think that it's still better and safer to extend `struct ceph_mds_request_head` I'll do. Just confirm, please.

Yeah, please. Or it will be hard to maintain in future.

Reply:
I'm with @lxbsz on this one.

Reply:
Have done. Thanks!